bin/interactions/fun/xkcd.dart

```diff
@@ -0,0 +1,110 @@
+import 'dart:async';
+
+import 'package:dio/dio.dart';
+import 'package:nyxx/nyxx.dart';
+import 'package:nyxx_interactions/interactions.dart';
+
+import '../../obsidian_dart.dart';
+import '../../utils/constants.dart';
+import '../../utils/embed.dart';
+
+class FunXkcdInteractions {
+  final BASE_URL = 'https://xkcd.com';
+
+  FunXkcdInteractions() {
+    botInteractions.registerSlashCommand(
+      SlashCommandBuilder(
+        'xkcd',
+        'XKCD group of commands.',
+        [
+          CommandOptionBuilder(CommandOptionType.subCommand, 'latest',
+              'Retrives lastest XKCD comic.')
+            ..registerHandler(xkcdLatestSlashCommand),
+          CommandOptionBuilder(
+            CommandOptionType.subCommand,
+            'search',
+            'Retrives XKCD comic.',
+            options: [
+              CommandOptionBuilder(
+                  CommandOptionType.integer, 'comic', 'XKCD Comic number.',
+                  required: true)
+            ],
+          )..registerHandler(xkcdComicSlashCommand),
+        ],
+      ),
+    );
+  }
+
+  EmbedBuilder xkcdEmbed(SlashCommandInteractionEvent event, String title,
+      String desc, String imgUrl, String footerText) {
+    return EmbedBuilder()
+      ..title = title
+      ..description = desc
+      ..color = DiscordColor.sapGreen
+      ..imageUrl = imgUrl
+      ..addFooter((footer) {
+        footer.text = footerText;
+        footer.iconUrl = event.interaction.userAuthor?.avatarURL();
+      });
+  }
+
+  Future<dynamic> xkcdLatestComicInfo() async {
```

Comment:
Instead of `Future<dynamic>` you can add `Future<String>` and return `response.data.toString()` although this is your choice.

Reply:
`response.data` was much easier to work with.

Reply:
Alright